### DISCUSSION TITLE
Object Detection Coordinate Tracking to Ignore Redundant Detections

### DISCUSSION THREAD

**Discussion by madman2012** [roflcoopter/viseron#789] [Ideas]
Hi Viseron Team,

First, I want to express my gratitude for your hard work and dedication in developing Viseron. It is a fantastic tool that has significantly improved my home surveillance setup.

I would like to propose a new feature that I believe would make it unique and enhance the efficiency of the object detection module in Viseron. 

The idea is to implement a mechanism to track the coordinates of detected objects and ignore repeated detections of objects that remain in the same location until they move. The require_motion key has not functioned in my testing.

Scope of the Feature

Background:
In many monitoring scenarios, objects such as cars or static objects are repeatedly detected at the same coordinates over time. This leads to redundant logs and unnecessary processing. By ignoring these repeated detections, we can improve the system's performance and clarity.

Proposed Solution:

Object State Tracking: Maintain a dictionary to track detected objects and their coordinates along with a timestamp.
Coordinate Comparison: Compare the coordinates of new detections with previous ones to determine if the object has moved significantly.

Threshold and Timeout Parameters: Introduce configurable parameters for movement threshold and timeout.
Update and Cleanup: Update the tracking dictionary with new coordinates and periodically clean up old entries.

Configuration:
Add the following options to the YAML configuration:
object_tracker:
  threshold: 0.01
  timeout: 10

Benefits:
Reduces redundant processing of stationary objects.
Improves log clarity by minimizing repeated detections.
Enhances the efficiency and performance of the object detection module.

Example Usage:
The feature would be integrated into the existing object detection pipeline, filtering out redundant detections before further processing.

I believe this feature would greatly benefit many users and improve the overall functionality of Viseron. I hope the community and the development team will consider it for a future release.

Thank you for considering this request. I am happy to provide further details or assist with testing if needed.

Best regards,

**Comment by roflcoopter** [comment]
Great idea and thank you for such a well written and thought out feature request.
Yes `require_motion` seems to be broken, there is an open issue on this that i have not been able to resolve yet.

I cant make any promises when I can implement this, I am pretty much alone in working on Viseron as of now and i am working hard to get v3 released.

My plan for future releases as of now is:
- 3.0: 24/7 recordings
- 3.1 Live streaming
- 3.2: Unsure what to plan here

So depending on when 3.0 is released and how hard 3.1 is it might take a while before i get to this one sadly

**Comment by madman2012** [comment]
Great thank you for your consideration! Wish I could help. I am a hobbyist and self-learning basic programing. Getting into this after a 20yr career in financial services. If there is a way for me to help - happy to be of service.

________________________________

**Reply by roflcoopter** [reply]
I am actually self-taught myself! I work with programming but with much older programming tools than what is used here.
I knew nothing of Python, Docker, React etc when i started with Viseron.

**Comment by yangminglong** [comment]
If object detection coordinates can be obtained, can [PTZ](https://github.com/roflcoopter/viseron/pull/788#issue-2436450659)   be used to achieve automatic centering of objects ?
